### PR TITLE
Use stellar.accounts for OwnAccountLookupCache

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: go/stellar/global.go [hunk 1]
```diff
@@ -686,6 +686,26 @@ func (s *Stellar) OwnAccountCached(mctx libkb.MetaContext, accountID stellar1.Ac
...
+				return true, acc.IsPrimary, acc.Name, nil
+			}
+		}
+		s.accountsLock.Unlock()
+		return false, false, "", nil
+	}
+	s.accountsLock.Unlock()
+	return OwnAccountPlusName(mctx, accountID)
+}
+
```
Comment (mlsteele) [inline go/stellar/global.go]: dedup by having `OwnAccountCached` call `OwnAccountPlusNameCached` and discard the name.

File: go/stellar/stellar.go [hunk 2]
```diff
@@ -287,10 +287,27 @@ func OwnAccount(mctx libkb.MetaContext, accountID stellar1.AccountID) (own, isPr
...
+	if err != nil {
+		return false, false, "", err
+	}
+	for _, account := range bundle.Accounts {
+		if account.AccountID.Eq(accountID) {
+			return true, account.IsPrimary, account.Name, nil
+		}
+	}
+	return false, false, "", nil
+}
```
Comment (mlsteele) [inline go/stellar/stellar.go]: same dedup: `OwnAccount` should use `OwnAccountPlusName`.

File: go/stellar/stellarsvc/frontend_test.go [hunk 3]
```diff
@@ -1033,12 +1033,14 @@ func TestGetPaymentsLocal(t *testing.T) {
 	require.Equal(t, "", p.ToUsername)
 	require.Equal(t, "", p.ToAssertion)
 
+	err = tcs[1].Srv.walletState.RefreshAll(tcs[1].MetaContext(), "test")
+	require.NoError(t, err)
```
Comment (mlsteele) [inline go/stellar/stellarsvc/frontend_test.go]: was this added because there's a new race that could affect real apps?

File: go/stellar/util.go [hunk 4]
```diff
@@ -52,52 +50,20 @@ type OwnAccountLookupCache interface {
...
-	sync.RWMutex
-	loadErr  error
-	accounts map[stellar1.AccountID]*string
+type ownAccountLookupCacheFromGlobal struct {
+	libkb.Contextified
 }
 
-// NewOwnAccountLookupCache fetches the list of accounts in the background and stores them.
-// Was created before Stellar.accounts, and could probably benefit from using that cache.
 func NewOwnAccountLookupCache(mctx libkb.MetaContext) OwnAccountLookupCache {
```
Comment (mlsteele) [inline go/stellar/util.go]: Add a comment like
// `NewOwnAccountLookupCache` was obsoleted and exists only as an interface bridge. Feel free to remove it.

File: go/stellar/stellarsvc/frontend_test.go (hunk 3 above)
Comment (patrickxb) [reply go/stellar/stellarsvc/frontend_test.go]: I fixed it in ImportSecretKey (and removed these lines)

**Comment by patrickxb** [comment]
PTAL

**Pull request merged**